src/main/java/edu/dbsleipzig/stream/grouping/impl/algorithm/TableGroupingBase.java

```diff
@@ -614,34 +613,40 @@ protected Expression[] buildSelectGroupedVerticesGroupAttributes(){
     protected Expression buildJoinConditionForExpandedVertices(){
         PlannerExpressionSeqBuilder attributeJoinConditions = new PlannerExpressionSeqBuilder(getTableEnv());
         PlannerExpressionSeqBuilder temporalJoinConditions = new PlannerExpressionSeqBuilder(getTableEnv());
-        for (String key : vertexGroupingPropertyFieldNames.keySet()) {
-            attributeJoinConditions.expression($(vertexGroupingPropertyFieldNames.get(key))
-              .isEqual($(vertexAfterGroupingPropertyFieldNames.get(key))));
-            attributeJoinConditions.expression($(vertexGroupingPropertyFieldNames.get(key)).isNull()
-              .and($(vertexAfterGroupingPropertyFieldNames.get(key)).isNull()));
+        if (vertexGroupingPropertyKeys.size() > 0) {
+            for (String key : vertexGroupingPropertyFieldNames.keySet()) {
+                attributeJoinConditions.expression($(vertexGroupingPropertyFieldNames.get(key))
+                        .isEqual($(vertexAfterGroupingPropertyFieldNames.get(key))));
+                attributeJoinConditions.expression($(vertexGroupingPropertyFieldNames.get(key)).isNull()
+                        .and($(vertexAfterGroupingPropertyFieldNames.get(key)).isNull()));
+            }
         }
         if (useVertexLabels) {
             attributeJoinConditions.expression($(FIELD_VERTEX_LABEL).isEqual($(FIELD_SUPER_VERTEX_LABEL)));
             attributeJoinConditions.expression($(FIELD_VERTEX_LABEL).isNull().and($(FIELD_SUPER_VERTEX_LABEL)
               .isNull()));
         }
+
         temporalJoinConditions.expression($("preparedVerticesTime").isLessOrEqual($(FIELD_SUPER_VERTEX_ROWTIME)))
           .and($("preparedVerticesTime").isGreater($(FIELD_SUPER_VERTEX_ROWTIME).minus(windowConfig.getWindowExpression())));
         Expression[] joinConditionArray = attributeJoinConditions.build();
         ArrayList<ApiExpression> orConnectedConditions = new ArrayList<>();
 
-        for (int i=0; i<attributeJoinConditions.build().length-1; i+=2) {
-            orConnectedConditions.add(((ApiExpression) joinConditionArray[0]).or(joinConditionArray[1]));
+        for (int i=0; i<attributeJoinConditions.build().length; i+=2) {
```

Comment:
Can we discuss this? These lines need urgent some comments - otherwise we can not follow what's happening why